crypto/fipsmodule/rand/urandom.c

```diff
@@ -399,9 +399,26 @@ static int fill_with_entropy(uint8_t *out, size_t len, int block, int seed) {
   }
 #endif
 
+#if defined(OPENSSL_MACOS)
+  // POSIX 2024 says unistd, but man page for macos says the former
+  #include <sys/random.h>
+  // To get system randomness on MacOS and iOS we use |CCRandomGenerateBytes|
+  // rather than |getentropy| and /dev/urandom.
+  // TODO at most can do 256 bytes
+  if (getentropy(out, len) == 0) {
+    return 1;
+  } else {
```

Comment:
This is inside the function `fill_with_entropy`, which is used for [both `CRYPTO_sysrand` and `CRYPTO_sysrand_for_seed`](https://github.com/aws/aws-lc/blob/d726d0616bc30029bf6e7bada9f0a2ba3a7e3b00/crypto/fipsmodule/rand/urandom.c#L475-L487).  

`CCRandomGenerateBytes`/`arc4random_buf` provide the output of a PRNG and are better suited for `CRYPTO_sysrand`, and `getentropy` is intended for seeding aPRNG and is better suited for `CRYPTO_sysrand_for_seed`.

On MacOS/OpenBSD, I think we should continue to use `CCRandomGenerateBytes`/`arc4random_buf` when `seed == 0` and only use `getentropy` when `seed == 1`.  How important is the distinction of `CRYPTO_sysrand` and `CRYPTO_sysrand_for_seed`?

Reply:
this is refactored now.